sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -248,7 +248,11 @@ case class ScalarSubquery(
     children: Seq[Expression] = Seq.empty,
     exprId: ExprId = NamedExpression.newExprId)
   extends SubqueryExpression(plan, children, exprId) with Unevaluable {
-  override def dataType: DataType = plan.schema.fields.head.dataType
+  override def dataType: DataType = if (!plan.schema.fields.isEmpty) {
+    plan.schema.fields.head.dataType
+  } else {
+    NullType
```

Comment:
Can we pull out when this can happen? Is it test only issue or dose it happen in an actual end-to-end case?

Reply:
scalar subquery should throw analysisException if no column is passed, but in this case it is throwing NoSuchElementException, so there is no functional issue, it's just an testcase issue which is using boundary condition

Reply:
Thanks. Can you pull out which test uses empty child against scalar subquery, and describe in PR description?